### PR TITLE
Capture SeatNonBids for rejected bids due to floor enforcement

### PR DESCRIPTION
**Pull request by ashishshinde-pubm**

### CODE REVIEW

**Review by pm-jaydeep-mohite — APPROVED** [review]
LGTM

File: exchange/exchangetest/floors_enforcement.json [hunk 1]
```diff
@@ -151,6 +151,35 @@
...
         ]
+      },
+      "ext": {
+        "prebid": {
+          "seatnonbid": [
+           {
+            "nonbid": [
+              {
+                "impid": "my-imp-id",
+                "statuscode": 301,
```
Comment (AlexBVolcy) [inline exchange/exchangetest/floors_enforcement.json:162]: Can you add another json test for the new 304 status code?

File: exchange/exchange.go [hunk 2]
```diff
@@ -399,6 +399,11 @@ func (e *exchange) HoldAuction(ctx context.Context, r *AuctionRequest, debugLog
 				errs = append(errs, &errortypes.Warning{
 					Message:     fmt.Sprintf("%s bid id %s rejected - bid price %.4f %s is less than bid floor %.4f %s for imp %s", rejectedBid.Seat, rejectedBid.Bids[0].Bid.ID, rejectedBid.Bids[0].Bid.Price, rejectedBid.Currency, rejectedBid.Bids[0].BidFloors.FloorValue, rejectedBid.Bids[0].BidFloors.FloorCurrency, rejectedBid.Bids[0].Bid.ImpID),
 					WarningCode: errortypes.FloorBidRejectionWarningCode})
+				rejectionReason := ResponseRejectedBelowFloor
+				if rejectedBid.Bids[0].Bid.DealID != "" {
+					rejectionReason = ResponseRejectedBelowDealFloor
```
Comment (AlexBVolcy) [inline exchange/exchange.go:404]: Can you add a unit test to cover this line?

Comment (ashishshinde-pubm) [reply exchange/exchange.go:404]: got covered through json test.

File: exchange/exchangetest/floors_enforcement.json (hunk 1 above)
Comment (ashishshinde-pubm) [reply exchange/exchangetest/floors_enforcement.json:162]: yes, done.

**Review by AlexBVolcy — APPROVED** [review]

**Review by guscarreon — APPROVED** [review]
LGTM

**Pull request merged**